python/samples/demos/process_with_dapr/app.py

```diff
@@ -0,0 +1,194 @@
+
+
+# Define a sample `BStep` step that will emit an event after 2 seconds.
+# The event will be sent to the `CStep` step with the data `I did B`.
+class BStep(KernelProcessStep):
+    @kernel_function()
+    async def do_it(self, context: KernelProcessStepContext):
+        print("##### BStep ran.")
+        await asyncio.sleep(2)
+        await context.emit_event(process_event=CommonEvents.BStepDone.value, data="I did B")
+
+
+# Define a sample `CStepState` that will keep track of the current cycle.
+class CStepState(KernelBaseModel):
+    current_cycle: int = 1
+
+
+# Define a sample `CStep` step that will emit an `ExitRequested` event after 3 cycles.
+class CStep(KernelProcessStep[CStepState]):
+    state: CStepState = Field(default_factory=CStepState)
+
+    # The activate method overrides the base class method to set the state in the step.
+    async def activate(self, state: KernelProcessStepState[CStepState]):
+        """Activates the step and sets the state."""
+        step_state = CStepState.model_validate(state.state)
+        print(f"##### CStep activated with Cycle = '{step_state.current_cycle}'.")
+        self.state = state.state
+
+    @kernel_function()
+    async def do_it(self, context: KernelProcessStepContext, astepdata: str, bstepdata: str):
+        self.state.current_cycle += 1
+        if self.state.current_cycle >= 3:
+            print("##### CStep run cycle 3 - exiting.")
+            await context.emit_event(process_event=CommonEvents.ExitRequested.value)
+            return
+        print(f"##### CStep run cycle {self.state.current_cycle}")
+        await context.emit_event(process_event=CommonEvents.CStepDone.value)
+
+
+kernel = Kernel()
+
+
+def get_process() -> "KernelProcess":
+    # Define the process builder
+    process = ProcessBuilder(name="ProcessWithDapr")
+
+    # Add the step types to the builder
+    kickoff_step = process.add_step(step_type=KickOffStep)
+    myAStep = process.add_step(step_type=AStep)
+    myBStep = process.add_step(step_type=BStep)
+
+    # Intialize the CStep with an initial state and the state's current cycle set to 1
+    myCStep = process.add_step(step_type=CStep, initial_state=CStepState(current_cycle=1))
+
+    # Define the input event and where to send it to
+    process.on_input_event(event_id=CommonEvents.StartProcess.value).send_event_to(target=kickoff_step)
+
+    # Define the process flow
+    kickoff_step.on_event(event_id=CommonEvents.StartARequested.value).send_event_to(target=myAStep)
+    kickoff_step.on_event(event_id=CommonEvents.StartBRequested.value).send_event_to(target=myBStep)
+    myAStep.on_event(event_id=CommonEvents.AStepDone.value).send_event_to(target=myCStep, parameter_name="astepdata")
+
+    # Define the fan in behavior once both AStep and BStep are done
+    myBStep.on_event(event_id=CommonEvents.BStepDone.value).send_event_to(target=myCStep, parameter_name="bstepdata")
```

Comment:
Sure. The enum events are only an example, but I can override the __str__ method in the enum class so we can omit `.value`.

Reply:
Actually, overriding the __str__ method isn't going to work. If we don't pass in `.value` we will get the enum only. Then we'd need a bunch of `str(...)` conversion everywhere in the code...

Reply:
Since both the built-in events and probably many other use Enums, I think something like this is worth it:
```python
def on_event(self, event_id: str | Enum) -> "ProcessStepEdgeBuilder":
    if isinstance(event_id, Enum):
         event_id = event_id.value
```

Reply:
done